### PR TITLE
Add decimal to maxtype inference

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by llllllllll** [comment]
Should we just put `Decimal` in the `numeric` typeset?

**Comment by cpcloud** [comment]
We could do that, but we'd have to make `TypeSet`s know about inheritance relationships for parameterized types

**Comment by cpcloud** [comment]
Also, the maximum type is computed via `itemsize`, and I'm not sure what that means for a decimal.

File: datashape/typesets.py [hunk 1]
```diff
@@ -179,15 +179,24 @@ def maxtype(measure):
...
     if isoption:
         measure = measure.ty
-    assert matches_typeset(measure, scalar), 'measure must be numeric'
+    if not matches_typeset(measure, scalar) and not isinstance(measure, Decimal):
+        raise TypeError('measure must be numeric')
 
     if measure == bool_:
         result = bool_
-    else:
+    elif not isinstance(measure, Decimal):
```
Comment (llllllllll) [inline datashape/typesets.py]: can we flip this with the else block so that the `else` is the catch-all case

**Comment by llllllllll** [comment]
Okay, that makes sense

File: datashape/typesets.py (hunk 1 above)
Comment (cpcloud) [inline datashape/typesets.py]: yep that's clearler

**Pull request merged**